job-webhook/job_scraper.py

```diff
@@ -0,0 +1,353 @@
+            return False
+    
+    def run_once(self):
+        """Run a single scraping cycle."""
+        logger.info("Starting job scraping cycle...")
+        
+        # Get current page content
+        html_content = self.get_page_content()
+        if not html_content:
+            logger.error("Failed to fetch page content")
+            return
+        
+        # Check if content has changed
+        current_hash = self.get_content_hash(html_content)
+        cache = self.load_cache()
+        
+        if current_hash == cache.get("content_hash"):
+            logger.info("No changes detected in page content")
+            return
+        
+        # Parse current jobs
+        current_jobs = self.parse_job_table(html_content)
+        if not current_jobs:
+            logger.warning("No jobs found in current scrape")
+            return
+        
+        # Find new jobs
+        new_jobs = self.find_new_jobs(current_jobs, cache.get("jobs", []))
+        
+        if new_jobs:
+            logger.info(f"Found {len(new_jobs)} new job postings")
+            
+            # Send notifications for new jobs
+            for job in new_jobs:
+                success = self.send_discord_notification(job)
+                if success:
+                    time.sleep(2)  # Rate limit: wait 2 seconds between messages
+                else:
+                    logger.warning(f"Failed to send notification for "
+                                 f"{job.company} - {job.role}")
+        else:
+            logger.info("No new jobs found")
+        
+        # Update cache
+        self.save_cache(current_hash, current_jobs)
+        logger.info("Scraping cycle completed")
+    
+    def run_continuously(self, interval_minutes: int = 15):
+        """Run the scraper continuously with specified interval."""
+        logger.info(f"Starting continuous job scraping every "
+                   f"{interval_minutes} minutes")
+        
+        while True:
+            try:
+                self.run_once()
+                logger.info(f"Sleeping for {interval_minutes} minutes...")
+                time.sleep(interval_minutes * 60)
+                
+            except KeyboardInterrupt:
+                logger.info("Scraper stopped by user")
+                break
+            except Exception as e:
+                logger.error(f"Unexpected error in main loop: {e}")
+                logger.info("Continuing after error...")
+                time.sleep(60)  # Wait 1 minute before retrying
+
+def main():
+    """Main function to run the job scraper."""
+    # Configuration from environment variables
+    # Load environment variables from .env file if exists
+    webhook_url = os.getenv("DISCORD_WEBHOOK_URL")
+    github_url = os.getenv("GITHUB_URL", 
+                          "https://github.com/vanshb03/Summer2026-Internships")
+    scrape_interval = int(os.getenv("SCRAPE_INTERVAL", "15"))  # minutes
```

Comment:
Again, when you feel the need to add a comment to a variable, that usually means you didn't name it well enough. I see you already named the `run_continuously` parameter `interval_minutes`, so just follow the same spirit here.

```suggestion
    scrape_interval_minutes = int(os.getenv("SCRAPE_INTERVAL", "15"))
```